pkg/server/list_users_test.go

```diff
@@ -0,0 +1,529 @@
+package server
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	language "github.com/openfga/language/pkg/go/transformer"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	mockstorage "github.com/openfga/openfga/internal/mocks"
+	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/memory"
+	"github.com/openfga/openfga/pkg/storage/test"
+	"github.com/openfga/openfga/pkg/testutils"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func TestListUsersValidation(t *testing.T) {
```

Comment:
Note that the test cases contained herein are the same physical structure as those in the `functional_test.go` file with the caveat that we're testing the `Server` struct directly instead of through the gRPC client. It would maybe be nice to share the test cases (not the test harness) across these, putting them in the shared `pkg/server/test` package and importing them here in `list_users_test.go` and in `functional_test.go`.

Just an idea :shrug?

Reply:
Follow up improvement to our test composition. Doesn't need to block this PR.